pyproject.toml

```diff
@@ -24,7 +24,8 @@ classifiers = [
   "License :: OSI Approved :: Apache Software License"
 ]
 requires = [
-  "backports.functools-lru-cache; python_version < '3.3'"
+  "backports.functools-lru-cache; python_version < '3.3'",
+  "typing_extensions; python_version < '3.8'"
```

Comment:
If we're only importing while `TYPE_CHECKING` is true I don't think we need this dependency. It'll be supplied by the type checking tool.

Reply:
I added it because mypy was complaining about the lack of `LITERAL` and `UNION`, so I wondered if adding it explicitly would fix that issue.

It turned into other issues, unfortunately. Still working on it. May end up just ignoring the error on that line, because the hoops are getting a little ridiculous.